httomo/runner/loader.py

```diff
@@ -151,6 +151,97 @@ class UserDefinedAngles(NamedTuple):
 
 AnglesConfig: TypeAlias = Union[RawAngles, UserDefinedAngles]
 
+class PreviewDimConfig(NamedTuple):
+    start: int
+    stop: int
+
+class PreviewConfig(NamedTuple):
+    angles: PreviewDimConfig
+    detector_y: PreviewDimConfig
+    detector_x: PreviewDimConfig
+
+class Preview:
+    def __init__(
+        self,
+        preview_config: PreviewConfig,
+        dataset: h5py.Dataset,
+        image_key: Optional[h5py.Dataset],
+    ) -> None:
+        self.config = preview_config
+        self._dataset = dataset
+        self._image_key = image_key
+        self._check_within_data_bounds()
+        self._data_indices: Optional[List[int]] = None
+        self._global_shape: Optional[Tuple[int, int, int]] = None
+
+    def _check_within_data_bounds(self) -> None:
+        shape = self._dataset.shape
+        if self.config.angles.stop > shape[0]:
+            raise ValueError(
+                f"Preview indices in angles dim exceed bounds of data: "
+                f"start={self.config.angles.start}, stop={self.config.angles.stop}"
+            )
+
+        if self.config.detector_y.stop > shape[1]:
+            raise ValueError(
+                f"Preview indices in det y dim exceed bounds of data: "
+                f"start={self.config.detector_y.start}, "
+                f"stop={self.config.detector_y.stop}"
+            )
+
+        if self.config.detector_x.stop > shape[2]:
+            raise ValueError(
+                f"Preview indices in det x dim exceed bounds of data: "
+                f"start={self.config.detector_x.start}, "
+                f"stop={self.config.detector_x.stop}"
+            )
```

Comment:
can you add checks for `start >= stop` as well?

Reply:
Good idea, done in [ca9a0fa](https://github.com/DiamondLightSource/httomo/pull/209/commits/ca9a0fac3dddbfca0dc0578ad32d6158be54c713)